app/controllers/products_controller.rb

```diff
@@ -86,7 +86,8 @@ def completed?
     end
 
     def check_permission!
-      unless admin_login? || mentor_login? || adviser_login? || my_product? || completed?
+      practice = find_practice
+      unless admin_login? || mentor_login? || adviser_login? || my_product? || completed? || practice.open_product?
```

Comment:
```suggestion
      unless admin_login? || mentor_login? || adviser_login? || my_product? || completed? || find_practice.open_product?
```

の方がいいですね。

2つ利点があります。

- パフォーマンス
    - admin_login?などで先に条件がunless trueになったら`find_practice を実行しない分早くなります。
- 可読性
    - 変数は少ない方が読みやすく、変数にしても可読性が上がらないなら変数は少ない方がよいです。

Reply:
毎回find_practiceが走らないよう、改善しました！